src/lib/storage/frame_of_reference/frame_of_reference_iterable.hpp

```diff
@@ -0,0 +1,139 @@
+          _index_within_frame{0u},
+          _chunk_offset{0u} {}
+
+    // End iterator
+    explicit Iterator(OffsetValueIteratorT offset_value_it) : Iterator{{}, offset_value_it, {}} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    void increment() {
+      ++_offset_value_it;
+      ++_null_value_it;
+      ++_index_within_frame;
+      ++_chunk_offset;
+
+      if (_index_within_frame >= FrameOfReferenceColumn<T>::block_size) {
+        _index_within_frame = 0u;
+        ++_block_minimum_it;
+      }
+    }
+
+    bool equal(const Iterator& other) const { return _offset_value_it == other._offset_value_it; }
+
+    ColumnIteratorValue<T> dereference() const {
+      const auto value = static_cast<T>(*_offset_value_it) + *_block_minimum_it;
+      return ColumnIteratorValue<T>{value, *_null_value_it, _chunk_offset};
+    }
+
+   private:
+    ReferenceFrameIterator _block_minimum_it;
+    OffsetValueIteratorT _offset_value_it;
+    NullValueIterator _null_value_it;
+    size_t _index_within_frame;
+    ChunkOffset _chunk_offset;
+  };
+
+  template <typename OffsetValueDecompressorT>
+  class PointAccessIterator
+      : public BasePointAccessColumnIterator<PointAccessIterator<OffsetValueDecompressorT>, ColumnIteratorValue<T>> {
+   public:
+    // Begin Iterator
+    PointAccessIterator(const pmr_vector<T>* block_minima, const pmr_vector<bool>* null_values,
+                        OffsetValueDecompressorT* attribute_decoder, ChunkOffsetsIterator chunk_offsets_it)
+        : BasePointAccessColumnIterator<PointAccessIterator<OffsetValueDecompressorT>,
+                                        ColumnIteratorValue<T>>{chunk_offsets_it},
+          _block_minima{block_minima},
+          _null_values{null_values},
+          _attribute_decoder{attribute_decoder} {}
+
+    // End Iterator
+    explicit PointAccessIterator(ChunkOffsetsIterator chunk_offsets_it)
+        : PointAccessIterator{nullptr, nullptr, nullptr, chunk_offsets_it} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    ColumnIteratorValue<T> dereference() const {
+      const auto& chunk_offsets = this->chunk_offsets();
+
+      static constexpr auto block_size = FrameOfReferenceColumn<T>::block_size;
+
+      const auto is_null = (*_null_values)[chunk_offsets.into_referenced];
+      const auto block_minimum = (*_block_minima)[chunk_offsets.into_referenced / block_size];
+      const auto offset_value = _attribute_decoder->get(chunk_offsets.into_referenced);
+      const auto value = static_cast<T>(offset_value) + block_minimum;
+
+      return ColumnIteratorValue<T>{value, is_null, chunk_offsets.into_referencing};
+    }
+
+   private:
+    const pmr_vector<T>* _block_minima;
+    const pmr_vector<bool>* _null_values;
+    OffsetValueDecompressorT* _attribute_decoder;
```

Comment:
why is this called _attribute_decoder? When I read "attribute" I think "attribute vector" as in DictionaryColumn